src/http/request_template.rs

```diff
@@ -215,6 +218,19 @@ impl TryFrom<Endpoint> for RequestTemplate {
     }
 }
 
+impl<Ctx: PathString + HasHeaders> CacheKey<Ctx> for RequestTemplate {
+    fn cache_key(&self, ctx: &Ctx) -> anyhow::Result<u64> {
+        let mut hasher = DefaultHasher::new();
+
+        self.method.hash(&mut hasher);
+
+        let url = self.create_url(ctx)?;
+        url.hash(&mut hasher);
```

Comment:
Add additional headers that are available on the RequestTemplate.
Add unit tests for hash.